### PR TITLE
Crop before resize

### PR DESCRIPTION
**Pull request by roborourke**
Built on top of the `crop` branch, fixes some issues witht he latest version of sharp and makes sure you can always crop regardless of resize

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,15 +44,35 @@ module.exports = function( region, bucket, key, args, callback ) {
 				image.quality( Math.min( Math.max( Number( args.quality ), 0 ), 100 ) )
 			}
 
-			// resize & crop
+			// crop (assumes crop data from original)
+			if ( args.crop ) {
+				var cropValues = typeof args.crop === 'string' ? args.crop.split( ',' ) : args.crop
```
Comment (joehoyle) [inline index.js]: Could we use `px` suffix, to disambiguate from percentages, like photon does in https://developer.wordpress.com/docs/photon/api/#crop?

Comment (roborourke) [inline index.js]: Yeah you already did that bit below in the map function :)

Comment (joehoyle) [inline index.js]: Oh my bad!

File: server.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ http.createServer( function( request, response ) {
 		return response.end()
 	}
 
-	tachyon( config.region, config.bucket, decodeURI( params.pathname.substr(1) ), params.query, function( err, data, info ) {
+	tachyon.s3( config.region, config.bucket, decodeURI( params.pathname.substr(1) ), params.query, function( err, data, info ) {
```
Comment (joehoyle) [inline server.js]: Do we also need to change the `lambda-server` here too?

Comment (roborourke) [inline server.js]: I believe that's already done in the `lambda-handler.js` - makes a call to `tachyon.s3()`

**Pull request merged**